### PR TITLE
Jm/ts host sig

### PR DESCRIPTION
**Pull request by jimmyaxod**
This implements typescript support for the runtime/host.

### CODE REVIEW

File: ts/src/runtime/module.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import { Instance } from "./instance";
+import { Runtime } from "./runtime";
+import { SFunction } from "./sfunction";
+
+export class Module<T extends Signature> {
+  private runtime: Runtime<T>;
+  public sfunction: SFunction<T>;
+
+  private waInstance: undefined | WebAssembly.Instance;
```
Comment (ShivanshVij) [inline ts/src/runtime/module.ts:27]: wasmInstance

File: ts/src/runtime/sfunction.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+*/
+
+import { v4 as uuidv4 } from 'uuid';
+import { ScaleFunc } from "@loopholelabs/scalefile";
+import { Signature } from "@loopholelabs/scale-signature";
+
+import { Instance } from "./instance";
+import { Pool } from "./pool";
+
+export class SFunction<T extends Signature> {
```
Comment (ShivanshVij) [inline ts/src/runtime/sfunction.ts:24]: Can you call it Func?

**Review by ShivanshVij — APPROVED** [review]
LGTM with minor changes

**Comment by jimmyaxod** [comment]
Tidied up and started as fresh branch / PR https://github.com/loopholelabs/scale/pull/29

**Pull request closed without merging**